psi4/driver/driver_findif.py

```diff
@@ -0,0 +1,799 @@
+         displacements, and the second number is the level determined at.
+    initialize_string : function
+         A function that returns the string to print to show the caller was entered.
+         The string is both caller-specific and dependent on values determined
+         in this function.
+    verbose : int
+         Set to 0 to silence extra print information, regardless of the print level.
+         Used so the information is printed only during geometry generation, and not
+         during the derivative computation as well.
+
+    :returns: *dict*
+        Miscellaneous information required by callers.
+    """
+
+    core.print_out("\n         ----------------------------------------------------------\n")
+    core.print_out("                                   FINDIF\n")
+    core.print_out("                     R. A. King and Jonathon Misiewicz\n")
+    core.print_out("         ---------------------------------------------------------\n\n")
+
+    print_lvl = core.get_option("FINDIF", "PRINT")
+    num_pts = core.get_option("FINDIF", "POINTS")
+    disp_size = core.get_option("FINDIF", "DISP_SIZE")
+
+    data = {"print_lvl": print_lvl, "num_pts": num_pts, "disp_size": disp_size}
+
+    if print_lvl:
+        core.print_out(initialize_string(data))
+
+    # Get settings for CdSalcList, then get the CdSalcList.
+    method_allowed_irreps = 0x1 if mode == "1_0" else 0xFF
+    t_project = not core.get_global_option("EXTERN") and (not core.get_global_option("PERTURB_H"))
+    # core.get_option returns an int, but CdSalcList expect a bool, so re-cast
+    r_project = t_project and bool(core.get_option("FINDIF", "FD_PROJECT"))
+    salc_list = core.CdSalcList(mol, method_allowed_irreps, t_project, r_project)
+
+    # Convention: A capital N at the start of variable means "number of"
+    Natom = mol.natom()
+    Nirrep = salc_list.nirrep()
+    Nsalc = salc_list.ncd()
+
+    if print_lvl and verbose:
+        core.print_out("\tNumber of atoms is {:d}.\n".format(Natom))
+        if method_allowed_irreps != 0x1:
+            core.print_out("\tNumber of irreps is {:d}.\n".format(Nirrep))
+        core.print_out("\tNumber of {!s}SALCs is {:d}.\n".format("" if method_allowed_irreps != 0x1 else "symmetric ",
+                                                                 Nsalc))
+        core.print_out("\tTranslations projected? {:d}. Rotations projected? {:d}.\n".format(t_project, r_project))
+
+    # TODO: Replace with a generator from a stencil to a set of points.
+    pts_dict = {
+        3: {
+            "sym": ((-1, ), (1, )),
+            "asym": ((-1, ), ),
+            "off": ((1, 1), (-1, -1))
+        },
+        5: {
+            "sym": ((-2, ), (-1, ), (1, ), (2, )),
+            "asym": ((-2, ), (-1, )),
+            "off": ((-1, -2), (-2, -1), (-1, -1), (1, -1), (-1, 1), (1, 1), (2, 1), (1, 2))
+        }
+    }
+
+    # We won't need disps for a while, but let's validate num_pts early.
+    try:
```

Comment:
Can we write this as `if num_pts not in pts_dict: raise...`? Slightly more canonical.

Reply:
I can, but I'll still need to define `disps` somewhere. Probably closer to when I actually need it.